posts/experiences_with_rust/index.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Experiences using Packit for a Rust executable Project
+# any date that can be parsed with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse
+date: 2023-10-10
+authors: amulhern
+---
+
+# Experiences using Packit for a Rust Executable Project
+
+"How absurdly simple!" I cried.
+
+"Quite so!" said he, a little nettled. "Every problem becomes very childish when once it is explained to you."
+
+- Arthur Conan Doyle, "The Adventure of the Dancing Men"
+
+We have planned for a while to use Packit to generate packages on copr
+on demand for our somewhat complicated Rust executable, stratisd. It
+looked like this was going to be challenging, and in a sense it was,
+but once the task was completed, it turned out to have been pretty
+straightforward.
+
+<!-- truncate -->
+
+The two primary parts of our Stratis project are stratisd, a big,
+fairly complicated Rust project which generates multiple distinct Rust
+executables and also installs scripts and configuration files and so
+forth, and stratis-cli, a relatively simple Python project which
+communicates with stratisd over the D-Bus. Our main purpose in getting
+on-demand packages of both these projects was not to test the
+packaging, although that is a nice thing to do, but to have a
+convenient way to do functional testing of the project, frequently
+generating a stratisd package from one stratisd pull request and a
+stratis-cli package from a matching stratis-cli pull request.
+
+The Packit team had already obliged us by preparing a Packit
+configuration file for our stratis-cli repo, we thus merely had to
+figure out how to do the same for stratisd. We already had some
+scripts which we could run by hand to make the srpms for both projects
+so that we could manually send them to copr so, as it turned out, all
+we really needed to do was figure out how to write a Packit
+configuration file for stratisd so that Packit would do these tasks
+for us. There follows a quick summary of the decisions we made to
+solve this problem. Note that our whole and only task was to figure
+out how to create the source artifacts and edit the spec file and
+to provide these in such a way that Packit could take over from there,
+build the srpm, and carry on to request the copr builds which provide
+the binary packages for testing.
+
+A Packit configuration is used to set certain configuration _options_,
+to override certain _actions_, or to define certain _hooks_. The
+available actions are specific to a given task or groups of tasks. For
+example, the srpm build task and the copr task, since they both build
+an srpm, will naturally share a few actions. If an action is not
```

Comment:
I am not completely sure how you mean this, since "SRPM task" is basically part of the Copr task = one `copr_build` job in the Packit config

Reply:
I'll smooth this out entirely. I may have picked this idea up from an obsolete example somewhere, but I if so, it's not important.